src/app/settings/categories/regressiontypes/regressiontypes.component.ts

```diff
@@ -99,25 +100,60 @@ export class RegressionTypesComponent implements OnInit, OnDestroy {
         }
         this._settingsservice
             .getEntities(this.configSettings.nssBaseURL + this.configSettings.regTypeURL+"?regions="+ this.selectedRegionID +"&statisticgroups="+ this.selectedStatisticID)
-            .subscribe(regs => {
-                this.regressionTypes = regs;
+            .subscribe(res => {
+                this.nssRegressionTypes = res;
+                nssReturn = true;
+                if (nssReturn == true && gsReturn == true) {
+                    this.combineRegressionTypes();
+                }
+            });
+        this._settingsservice
+            .getEntities(this.configSettings.gageStatsBaseURL + this.configSettings.regTypeURL+"?regions="+ this.selectedRegionID +"&statisticgroups="+ this.selectedStatisticID)
+            .subscribe(res => {
+                this.gsRegressionTypes = res;
+                gsReturn = true;
+                if (nssReturn == true && gsReturn == true) {
+                    this.combineRegressionTypes();
+                }
             });
-        
     }
 
     public onStatGroupSelect(e){
+        this._loaderService.showFullPageLoad();
         this.selectedStatistic = e;
         this.selectedStatisticID = e.id;
+        var nssReturn = false;
+        var gsReturn = false;
         if (e === 'none') {
             this.selectedStatisticID = "";
         } 
         if(this.selectedRegion === 'none'){
             this.selectedRegionID = "";
         }
-        this._settingsservice.getEntities(this.configSettings.nssBaseURL + this.configSettings.regTypeURL+"?regions="+ this.selectedRegionID +"&statisticgroups="+ this.selectedStatisticID).subscribe(res => {
+        this._settingsservice.getEntities(this.configSettings.nssBaseURL + this.configSettings.regTypeURL+"?regions="+ this.selectedRegionID +"&statisticgroups="+ this.selectedStatisticID)
+            .subscribe(res => {
             res.sort((a, b) => a.name.localeCompare(b.name));
-            this.regressionTypes = res;
+            this.nssRegressionTypes = res;
+            nssReturn = true;
+                if (nssReturn == true && gsReturn == true) {
+                    this.combineRegressionTypes();
+                }
         });
+        this._settingsservice.getEntities(this.configSettings.gageStatsBaseURL + this.configSettings.regTypeURL+"?regions="+ this.selectedRegionID +"&statisticgroups="+ this.selectedStatisticID)
+            .subscribe(res => {
+            res.sort((a, b) => a.name.localeCompare(b.name));
+            this.gsRegressionTypes = res;
+            gsReturn = true;
+            if (nssReturn == true && gsReturn == true) {
```

Comment:
Not worth changing, but you can just write `if (nssReturn && gsReturn)` since they're booleans